### ISSUE TITLE
Methods that return cell IDs should return Cell object

### ISSUE THREAD

**Issue by banesullivan** [pyvista/pyvista#3916]
### Describe the feature you would like to be added.

We have some great convienance methods like `find_closest_cell()` and `find_containing_cell()` and others. These return an integer Cell ID -- very expected. But we should return a `Cell` obect now that cells are well wrapped and supported in PyVista.

Currently, the code to get a cell looks like:

```py
cell_id = mesh.find_closest_cell([0,0,0])
cell = mesh.cell[cell_id]
```

I'm wondering if we could implement this with some sort of backward compatibility? Perhaps through an `__int__` method on the `Cell` class? I was hoping this would work for standard indexing, but maybe not?

```py
class Foo:
    def __init__(self, value: int):
        self.value = value
        
    def __int__(self):
        return self.value

...
>>> f = Foo(1)
>>> [0, 1, 2, 3][f]
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In [7], line 10
      6         return self.value
      8 f = Foo(1)
---> 10 [0, 1, 2, 3][f]

TypeError: list indices must be integers or slices, not Foo
```

### Pseudocode or Screenshots

Ideally, these methods would return a `Cell` object directly.

```py
cell = mesh.find_closest_cell([0,0,0])
```

**Comment by MatthewFlamm** [comment]
I'm conflicted here.  It does seem useful when you only want the Cell object.  But, I think one (maybe the most) common use of these methods would be using the id.  Also as you pointed out in your description, the Cell object has no information about the cell_id from the mesh. So to return only the Cell object, we have to patch something about the mesh, i.e. the cell id, to the Cell.  Unless there is some way to override `vtiGenericCell` that persists, we'd have to patch the id onto a concrete `Cell` object that has already been instantiated. 
 This starts to deviate PyVista from the VTK data model, but maybe not in a significant way.

I guess I'm saying that I'm +1 but only if it doesn't somehow interfere with using the cell_id or the vtk data model in some way.

**Comment by banesullivan** [comment]
Yeah, I didn't realize `vtkCell` does not track its own ID. I was hoping for a `GetID()` method like `GetPointIDs()`. We'd have to patch in support for tracking the ID of this cell on it parent (parent could be a mesh or another cell)... this gets messy.
